### ISSUE TITLE
Not downloading highest quality - HD not found

### ISSUE THREAD

**Issue by stordoff** [lightpohl/gb-dl#2]
Latest release, on Windows 10.

`gb-dl-win-x64.exe --api-key <clipped> --video-name "Quick Look: New Pokémon Snap"`

This downloads vf_ql_newpokemonsnap_050421_3200.mp4 (the 'high' quality), even though `"hd_url": "[...]vf_ql_newpokemonsnap_050421_4000.mp4"` is returned by the API. Attempting to force it results in:

`gb-dl-win-x64.exe --api-key <clipped> --video-name "Quick Look: New Pokémon Snap" --quality hd`
`quality hd not found for Quick Look: New Pokémon Snap`

There is also an even higher quality version available at vf_ql_newpokemonsnap_050421_8000.mp4, but this appears to be an API limitation.

**Comment by lightpohl** [comment]
@stordoff - Hey! Thanks for letting me know — let me see what's changed. I know some older videos are missing certain qualities, but I definitely want to support the latest videos. 👍

**Comment by lightpohl** [comment]
Unfortunately, this looks like a bug on the Giant Bomb API side of things. My current guess is that the request is currently cached on their end before the HD video was available — I'll check back tomorrow and open a bug report with them if it's still missing. 👍  

Here's the response currently being sent back for the video. Note the missing `hd_url` that should be included.

```js
{
  api_detail_url: 'https://www.giantbomb.com/api/video/2300-16752/',
  deck: 'Award winning nature photographers, Vinny and Jan, embark on a journey to document nothing but Bidoofs.',
  id: 16752,
  name: 'Quick Look: New Pokémon Snap',
  premium: false,
  publish_date: '2021-05-04 10:24:00',
  site_detail_url: 'https://www.giantbomb.com/videos/new-pokemon-snap/2300-16752/',
  video_show: {
    api_detail_url: 'https://www.giantbomb.com/api/video_show/2340-3/',
    id: 3,
    title: 'Quick Looks',
    position: 8,
    site_detail_url: 'https://www.giantbomb.com/shows/quick-looks/',
    image: {/* ... */},
    logo: {/* ... */},
  },
  low_url: 'https://static-giantbombvideo.cbsistatic.com/vr/2021/05/04/649816/vf_ql_newpokemonsnap_050421_1800.mp4',
  high_url: 'https://static-giantbombvideo.cbsistatic.com/vr/2021/05/04/649816/vf_ql_newpokemonsnap_050421_3200.mp4',
  resource_type: 'video'
}
```

**Comment by lightpohl** [comment]
@stordoff - This is a bit in the weeds, but I noticed something that's been [reported over on the Giant Bomb forums](https://www.giantbomb.com/forums/api-developers-3017/some-search-results-are-missing-the-video-url-fiel-1873852/). When passed a `--video-name`, `gb-dl` uses the "search" API to find the video (which is missing `hd_url`), but if you use the `--show-name` and `--video-number` it has a value for `hd_url`.

- `npx gb-dl --api-key <REDACTED> --show-name "Quick Looks" --video-number 0`
  - Includes `hd_url` property in response.
- `npx gb-dl --api-key <REDACTED> --video-name "Quick Look: New Pokémon Snap"`
  - Missing `hd_url` property in response.

**Comment by stordoff** [comment]
> I noticed something that's been [reported over on the Giant Bomb forums](https://www.giantbomb.com/forums/api-developers-3017/some-search-results-are-missing-the-video-url-fiel-1873852/).

Huh, that's a weird limitation/bug. I suppose a workaround would be to send an extra API request (using the returned `api_detail_url`) and pull the URLs from that, but that's not ideal when the API is rate limited.

**Comment by lightpohl** [comment]
@stordoff - Ah! That's really clever! `gb-dl` manually does a bit of rate limiting so one extra request shouldn't bog it down too much. Let me see if that works.

**Comment by lightpohl** [comment]
Worked like a charm! Thanks again. ❤️ 

Fixed in [v2.6.8](https://github.com/lightpohl/gb-dl/releases/tag/v2.6.8).
